controllers/storagecluster/job_templates.go

```diff
@@ -57,6 +58,20 @@ func (obj *ocsJobTemplates) ensureCreated(r *StorageClusterReconciler, sc *ocsv1
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return reconcile.Result{}, fmt.Errorf("failed to create Template : %v", err.Error())
 		}
+
+		// Delete a template job after the job gets completed
+		// This eliminates the need to delete the older/existing job when triggering the job for a second time
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, client.ObjectKey{Name: fmt.Sprintf("%s-job", template.Name), Namespace: template.Namespace}, job)
+		if err == nil {
+			if job.Status.Succeeded >= 1 {
+				err = r.Client.Delete(r.ctx, job)
+				if err != nil {
+					r.Log.Info("Job " + job.Name + " couldn't be deleted successfully after completion")
+				}
```

Comment:
Line 72 will be mistakenly logged if there is an error
```suggestion
				} else {
```